hippod/api_object_post.py

```diff
@@ -75,85 +82,185 @@ def create_container_data_merge_issue_new(app, sha_sum, object_item, submitter):
+        fd_minor = open(minor_file_path, 'w')
+        fd_minor.write(cd_minor)
+        fd_minor.close()
+        cd_minor = json.loads(cd_minor)
 
-def is_obj_already_in_db(app, sha_sum):
+def is_obj_major_already_in_db(app, sha_major):
     path = os.path.join(app['DB_OBJECT_PATH'],
-                        sha_sum[0:2],
-                        sha_sum,
+                        sha_major[0:2],
+                        sha_major,
                         'container.db')
     if os.path.isfile(path):
         return True
     else:
         return False
 
 
+def is_obj_minor_already_in_db(app, sha_major, sha_minor):
+    path = os.path.join(app['DB_OBJECT_PATH'],
+                        sha_major[0:2],
+                        sha_major,
+                        sha_minor,
+                        'subcontainer.db')
+    if os.path.isfile(path):
+        return True
+    else:
+        return False
 
 
-def write_cont_obj_by_id(app, sha, py_object):
+def write_cont_obj_by_id(app, sha_major, py_object):
     data = json.dumps(py_object, sort_keys=True,indent=4,
                       separators=(',', ': '))
     path = os.path.join(app['DB_OBJECT_PATH'],
-                        sha[0:2],
-                        sha,
+                        sha_major[0:2],
+                        sha_major,
                         'container.db')
     fd = open(path, 'w')
     fd.write(data)
     fd.close()
 
 
-def write_achievement_file(app, sha, id_no, achievement):
-    # swap out data if mime types say so and modify
-    # achievement inplace to reflect changes
-    hippod.mime_data_db.save_object_item_data_list(app, achievement)
-
-    path = os.path.join(app['DB_OBJECT_PATH'],
-                        sha[0:2],
-                        sha,
-                        'achievements',
-                        '{}.db'.format(id_no))
-    data = json.dumps(achievement, sort_keys=True,indent=4,
+def write_subcont_obj_by_id(app, sha_major, sha_minor, py_object):
+    data = json.dumps(py_object, sort_keys=True,indent=4,
                       separators=(',', ': '))
+    path = os.path.join(app['DB_OBJECT_PATH'],
+                        sha_major[0:2],
+                        sha_major,
+                        sha_minor,
+                        'subcontainer.db')
     fd = open(path, 'w')
     fd.write(data)
     fd.close()
 
 
-def write_attachment_file(app, sha, id_no, attachment):
-    path = os.path.join(app['DB_OBJECT_PATH'],
-                        sha[0:2],
-                        sha,
-                        'attachments',
-                        '{}.db'.format(id_no))
-    data = json.dumps(attachment, sort_keys=True,indent=4,
-                      separators=(',', ': '))
-    fd = open(path, 'w')
-    fd.write(data)
-    fd.close()
+def write_achievement_file(app, sha_major, sha_minor, id_no, achievement):
+    # swap out data if mime types say so and modify
+    # achievement inplace to reflect changes
+    try:
```

Comment:
mmh, the try catch is to coarse. It covers a large area of functionalty somehow. Errors should be catched where thrown. This seems something is hidden, catched and not corretly addresses. If required the concrete Error is required at except, not the general Exception. Mhh